### PR TITLE
Introduce Clock interface for time measurements

### PR DESCRIPTION
**Pull request by yurishkuro**
- Use nano-time for measuring duration
- Fix the tests, remove powermock dependency

- [ ] add more tests for duration calculation

### CODE REVIEW

**Comment by yurishkuro** [comment]
@adriancole we discussed in the past how this lib used `(millis*1000 + nanos/1000)`, I now changed it to be "more better".

File: jaeger-core/src/main/java/com/uber/jaeger/utils/SystemClock.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class SystemClock implements Clock {
+
+    @Override
+    public long currentTimeMicros() {
+        return System.currentTimeMillis() * 1000;
+    }
+
+    @Override
+    public long currentTimeNanos() {
```
Comment (adriancole) [inline jaeger-core/src/main/java/com/uber/jaeger/utils/SystemClock.java]: I'd probably not use the word "Time" here, eventhough nanoTime does. maybe tick or nanoTick? https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Ticker.html

Comment (adriancole) [inline jaeger-core/src/main/java/com/uber/jaeger/utils/SystemClock.java]: reason is that people might be confused that it relates to the other "time" in this interface which it doesn't

**Comment by adriancole** [comment]
naming bikeshed aside, looks good.

File: jaeger-context/src/test/java/com/uber/jaeger/context/CallableTest.java [hunk 2]
```diff
@@ -19,11 +19,8 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package com.uber.jaeger.filters.jaxrs2;
+package com.uber.jaeger.context;
```
Comment (yurishkuro) [inline jaeger-context/src/test/java/com/uber/jaeger/context/CallableTest.java]: @oibe these tests weren't in the proper tests dir, so not sure if they were even exercised previously (intelliJ didn't understand them as tests)

File: jaeger-context/src/test/java/com/uber/jaeger/context/TracedExecutorServiceTest.java [hunk 3]
```diff
@@ -41,22 +38,22 @@
...
-import static org.powermock.api.mockito.PowerMockito.when;
+import static org.mockito.Mockito.when;
 
 public class TracedExecutorServiceTest {
 
     TimeUnit TIME_UNIT = TimeUnit.MILLISECONDS;
     TracedExecutorService tracedExecutorService;
     ExecutorService wrappedExecutorService;
-    Tracer tracer = new Tracer.Builder("test-executor-service", new InMemoryReporter(), new ConstSampler(true)).build();
+    //Tracer tracer = new Tracer.Builder("test-executor-service", new InMemoryReporter(), new ConstSampler(true)).build();
```
Comment (yurishkuro) [inline jaeger-context/src/test/java/com/uber/jaeger/context/TracedExecutorServiceTest.java]: removing this

**Comment by oibe** [comment]
lgtm

File: jaeger-context/src/test/java/com/uber/jaeger/context/CallableTest.java (hunk 2 above)
Comment (oibe) [inline jaeger-context/src/test/java/com/uber/jaeger/context/CallableTest.java]: Noted.

**Pull request merged**